frontends/mit-learn/src/page-components/EcommerceFeature/EcommerceFeature.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react"
+import { useFeatureFlagEnabled } from "posthog-js/react"
+import { ForbiddenError } from "@/common/permissions"
+import { FeatureFlags } from "@/common/feature_flags"
+
+type EcommerceFeatureProps = {
+  children: React.ReactNode
+}
+
+/**
+ * Simple wrapper to standardize the feature flag check for ecommerce UI pages.
+ * If the flag is enabled, display the children; if not, throw a ForbiddenError
+ * like you'd get for an unauthenticated route.
+ *
+ * There's a PostHogFeature component that is provided but went this route
+ * because it seemed to be inconsistent - sometimes having the flag enabled
+ * resulted in it tossing to the error page.
+ *
+ * Set the feature flag here using the enum, and then make sure it's also
+ * defined in commmon/feature_flags too.
+ */
+
+const EcommerceFeature: React.FC<EcommerceFeatureProps> = ({ children }) => {
+  const ecommFlag = useFeatureFlagEnabled(FeatureFlags.EnableEcommerce)
+
+  if (ecommFlag === false) {
+    throw new ForbiddenError("Not enabled.")
+  }
```

Comment:
_Post-merge comment_

I noticed some odd behavior with PostHog and feature flags that I thought was worth mentioning. Essentially, I think there's no reliable way—with `useFeatureFlagEnabled`—to determine whether a flag is "off" or "not yet loaded". (_Note: the `useFeatureFlagEnabled` is really simple. Here's its [source](https://github.com/PostHog/posthog-js/blob/main/react/src/hooks/useFeatureFlagEnabled.ts)._)

`useFeatureFlagEnabled` returns `boolean | undefined`. This seems like the "normal" behavior:
1. `useFeatureFlagEnabled("flag-A")` returns undefined before any flags have loaded
2. When any flags have loaded, `useFeatureFlagEnabled("flag-A")` returns `true` if the flag is on, and `false` if the flag is off or has not loaded yet.
    - "When any flags have loaded" includes:
        - bootstrapped flags
        - flags determined via API call to posthog site

So the following is possible:

<img width="1003" alt="Screenshot 2024-10-04 at 12 30 41 PM" src="https://github.com/user-attachments/assets/e8bbe401-5106-4035-8667-2722ddf340bb">

In this scenario, we were attempting to:
1. If flags not loaded yet, return null
2. If flags loaded, either
    - redirect
    - or render the cart page 

But this doesn't seem to work quite right if any flags are bootstrapped, as shown in the sreenshot.